swarm/api/http/server.go

```diff
@@ -402,22 +401,33 @@ func resourcePostMode(path string) (isRaw bool, frequency uint64, err error) {
 func (s *Server) HandlePostResource(w http.ResponseWriter, r *Request) {
 	log.Debug("handle.post.resource", "ruid", r.ruid)
 	var err error
-	var addr storage.Address
-	var name string
+	var rootAddr storage.Address
 	var outdata []byte
-	isRaw, frequency, err := resourcePostMode(r.uri.Path)
+
+	// Creation and update must send mru.UpdateRequest JSON structure
+	body, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		Respond(w, r, err.Error(), http.StatusBadRequest)
+		Respond(w, r, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	updateRequest, err := mru.DecodeUpdateRequest(body) // decodes request JSON
+	if err != nil {
+		Respond(w, r, err.Error(), http.StatusBadRequest) //TODO: send different status response depending on error
 		return
 	}
 
-	// new mutable resource creation will always have a frequency field larger than 0
-	if frequency > 0 {
+	// Verify that the signature is intact and that the signer is authorized
+	// to update this resource
+	if err = updateRequest.Verify(); err != nil {
+		Respond(w, r, err.Error(), http.StatusForbidden)
+		return
+	}
 
-		name = r.uri.Addr
+	// new mutable resource creation will always have a frequency field larger than 0
+	if updateRequest.Frequency() > 0 {
```

Comment:
AFAICS outdata is only written if this is true so try to just return and get rid of the else